src/message-fragmentation.md

```diff
@@ -43,8 +49,12 @@ Note that:
 - Nodes should enforce a limit on the number of fragments per message (see the [parameters
   chapter](./parameters.md)), by simply discarding fragments with a too-large
   `num_fragments_minus_1`.
+// TODO should common parameter like this be part of the runtime api?
 - Multiple fragments with the same `message_id` and `fragment_index` may be received. It is
   recommended that only one be kept.
+// TODO maybe also acceptable to drop all if they differs in content.
+// Explicitelly this means that a packet can be send twice? Or we store only per message_id (in this
```

Comment:
Yes, packets may be sent multiple times (although as noted never with the same SURBs). See the stuff about retransmission in `requests-and-replies.md`. The message ID is reused when retransmitting to increase the chance that we get a complete message (if we were only missing a single fragment the first time, we only care about this fragment making it the second time).

Reply:
It think it should be stated somewhere explicitly (I mean it is in the spec, but could be good to get the info without reading the whole spec, like at the start of some part).
Sending packet multiple time is a choice of the user, but probably in some circumstances he do not want to do so (this leaks info).